lib/internal/webstreams/readablestream.js

```diff
@@ -1609,6 +1600,11 @@ function readableStreamFulfillReadRequest(stream, chunk, done) {
   } = stream[kState];
   assert(reader[kState].readRequests.length);
   const readRequest = ArrayPrototypeShift(reader[kState].readRequests);
+
+  // TODO(@jasnell): It's not clear under what exact conditions done
+  // will be true here. The spec requires this check but none of the
+  // WPT's or other tests trigger it. Will need to investigate how to
+  // get coverage for this.
```

Comment:
The reference implementation also has no coverage on these lines... It looks like this might be dead code in the specification. 🤔 

There are 3 places where the spec calls `ReadableStreamFulfillReadRequest`. However, 2 of those are from `Enqueue()` and always pass `done = false`, so they don't count.

The remaining place is [step 5.1 in `ReadableByteStreamControllerCommitPullIntoDescriptor`](https://streams.spec.whatwg.org/commit-snapshots/d0bd912c2c1547aabee61c950ff624e0937d8779/#ref-for-readable-stream-fulfill-read-request%E2%91%A0). This is called from 3 places:
1. [Step 2.3.2 of `ReadableByteStreamControllerProcessPullIntoDescriptorsUsingQueue`](https://streams.spec.whatwg.org/commit-snapshots/d0bd912c2c1547aabee61c950ff624e0937d8779/#ref-for-readable-byte-stream-controller-commit-pull-into-descriptor). However, this operation starts by asserting that `closeRequested = false`, which implies `state = 'readable'`. None of the steps in this operation change `state`, so by the time we reach 2.3.2 it'll still be `'readable'` and `done` will be set to `false`.
2. [Step 3.1.2 of `ReadableByteStreamControllerRespondInClosedState`](https://streams.spec.whatwg.org/commit-snapshots/d0bd912c2c1547aabee61c950ff624e0937d8779/#ref-for-readable-byte-stream-controller-commit-pull-into-descriptor%E2%91%A0). However, step 3 checks if the stream has a BYOB reader attached, so we'll always hit step 6.2 (`ReadableStreamFulfillReadIntoRequest`) instead of step 5.1 (`ReadableStreamFulfillReadRequest`) of `ReadableByteStreamControllerCommitPullIntoDescriptor`.
3. [Step 8 of `ReadableByteStreamControllerRespondInReadableState`](https://streams.spec.whatwg.org/commit-snapshots/d0bd912c2c1547aabee61c950ff624e0937d8779/#ref-for-readable-byte-stream-controller-commit-pull-into-descriptor%E2%91%A1). However, this is once again only called when `state = 'readable'`, and none of the preceding steps can change `state` to `'closed'`.

Maybe we should change the spec and turn this into an `assert(done === false)` or something... 😛

Reply:
Yeah I was thinking the same. I tried hard to find ways of triggering that path, even artificially, and couldn't ever hit it.